### DISCUSSION TITLE
Table of LWG issues

### DISCUSSION THREAD

**Discussion by davidhunter22** [microsoft/STL#2473] [Ideas] [answered]
Hi, I don't think this exists already at least not that I could find.

It woud be nice to have a single table of LWG issues and their current status in terms of the STL.
Currently you can search on the changelog to find ones that are done but it would be nice to have everything in one place. This would include LWG issues that have been accepted but not implemented.  I am thinking something like the libc++ https://libcxx.llvm.org/Status/Cxx20.html#library-working-group-issues-status

Actually the same applies to some extent to papers for C++ 23, there is this https://docs.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-170  but it's normally a bit.  You can look at https://github.com/microsoft/STL/projects/8out of date but again the information is a bit spread out.

I'm not sure if it's possible but if these tables did exist it would be nice if punters like me could submit PR's to them. I for one, and I'm sure others, would be willing to do some work to help keep them up to date.

**Comment by CaseyCarter** [comment]
What utility would such a table provide beyond the ability to see incomplete LWG issues (https://github.com/microsoft/STL/issues?q=is%3Aissue+is%3Aopen+label%3Alwg), completed LWG isues (https://github.com/microsoft/STL/issues?q=is%3Aissue+is%3Aclosed+label%3Alwg), or all LWG issues (https://github.com/microsoft/STL/issues?q=is%3Aissue+label%3Alwg)?

**Comment by CaseyCarter** [comment]
> You can look at https://github.com/microsoft/STL/projects/8 but again the information is a bit spread out.

I'm hoping that the new Projects system (currently in Beta) will help address this with different views of the same project data.

**Comment by StephanTLavavej** [comment, marked as answer]
We used to maintain such a table (in Excel, internally) for LWG issues, but stopped maintaining it since keeping it updated was a lot of work and we didn't seem to need to refer to the table ever. (We still maintain a table of C++17/20/23 feature status, but the GitHub tracking issues, Changelog, and Status Chart provide the same info in a different form; we'd like to make the table itself publicly visible but that's still a todo that I hope to have time for in the future).

The main value that tables seem to provide is a condensed view of what features/issues were implemented in what release, but this is almost as convenient to see in the Changelog, by simply reading the top of each version section (as we always list features, then LWG issues, in that order).

@TylerMSFT also looked into adding an LWG issue table to the official Microsoft Docs, but abandoned that attempt after similarly discovering that researching each row (for which version an issue was implemented in, whether there are any special considerations like being Standard mode dependent, etc.) is an incredibly time-consuming task.

**Comment by davidhunter22** [comment]
I just noticed this https://github.com/orgs/microsoft/projects/1152. This is awesome :-)

**Reply by StephanTLavavej** [reply]
Yep, created that just last night - you're welcome! :heart_eyes_cat:
